fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -1501,68 +1497,71 @@ private void processMvRelatedMeta() {
-        // for community, version schema is [int], and the int value must be positive
-        // for starrocks, version schema is [-1, int, int]
+    public long loadVersion(DataInputStream dis, long checksum) throws IOException {
+        // for new format, version schema is [starrocksMetaVersion], and the int value must be positive
+        // for old format, version schema is [-1, metaVersion, starrocksMetaVersion]
         // so we can check the first int to determine the version schema
         int flag = dis.readInt();
-        long newChecksum = checksum ^ flag;
+        checksum = checksum ^ flag;
+        int starrocksMetaVersion;
         if (flag < 0) {
-            int communityMetaVersion = dis.readInt();
-            if (communityMetaVersion > FeConstants.META_VERSION) {
-                LOG.error("invalid meta data version found, cat not bigger than FeConstants.meta_version."
-                                + "please update FeConstants.meta_version bigger or equal to {} and restart.",
-                        communityMetaVersion);
-                System.exit(-1);
-            }
-            newChecksum ^= communityMetaVersion;
-            MetaContext.get().setMetaVersion(communityMetaVersion);
-            int starrocksMetaVersion = dis.readInt();
-            if (starrocksMetaVersion > FeConstants.STARROCKS_META_VERSION) {
-                LOG.error("invalid meta data version found, cat not bigger than FeConstants.starrocks_meta_version."
-                                + "please update FeConstants.starrocks_meta_version bigger or equal to {} and restart.",
-                        starrocksMetaVersion);
-                System.exit(-1);
-            }
-            newChecksum ^= starrocksMetaVersion;
-            MetaContext.get().setStarRocksMetaVersion(starrocksMetaVersion);
+            checksum ^= dis.readInt();
+            starrocksMetaVersion = dis.readInt();
+            checksum ^= starrocksMetaVersion;
         } else {
-            // when flag is positive, this is community image structure
-            int metaVersion = flag;
-            if (metaVersion > FeConstants.META_VERSION) {
-                LOG.error("invalid meta data version found, cat not bigger than FeConstants.meta_version."
-                                + "please update FeConstants.meta_version bigger or equal to {} and restart.",
-                        metaVersion);
-                System.exit(-1);
-            }
-            MetaContext.get().setMetaVersion(metaVersion);
+            // when flag is positive, this is new version format
+            starrocksMetaVersion = flag;
         }
 
-        long replayedJournalId = dis.readLong();
-        newChecksum ^= replayedJournalId;
+        if (starrocksMetaVersion > FeConstants.STARROCKS_META_VERSION) {
+            LOG.error("invalid meta data version found, cat not bigger than FeConstants.starrocks_meta_version."
```

Comment:
typo, and the logging message is also not proper, if `starrocksMetaVersion > FeConstants.STARROCKS_META_VERSION`, it means that  we use an old version FE to load newer version image, what we should do is to use right(newer) version FE to restart, not just modify the code.

Reply:
does this `if` condition here means that we cannot rollback to older version, like from 3.1 to 3.0?

Reply:
changed